cmd/gardenlet/app/migration.go

```diff
@@ -153,3 +159,36 @@ func deleteOrphanedValiVPAs(ctx context.Context, log logr.Logger, c client.Clien
 
 	return flow.Parallel(fns...)(ctx)
 }
+
+// TODO(MichaelEischer): Remove this function after v1.101 has been released
+func createOSCHashMigrationSecret(ctx context.Context, seedClient client.Client) error {
+	namespaceList := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, namespaceList, client.MatchingLabels(map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot})); err != nil {
+		return err
+	}
+
+	var tasks []flow.TaskFn
+	for _, ns := range namespaceList.Items {
+		if ns.DeletionTimestamp != nil || ns.Status.Phase == corev1.NamespaceTerminating {
+			continue
+		}
+		namespace := ns
+		tasks = append(tasks, func(ctx context.Context) error {
+			if err := seedClient.Get(ctx, types.NamespacedName{Namespace: namespace.Name, Name: operatingsystemconfig.PoolHashesSecret}, &corev1.Secret{}); err == nil || !apierrors.IsNotFound(err) {
+				// nothing to do if secret already exists
+				return err
```

Comment:
You could use a better error message with `fmt.Errorf`

Reply:
I've wrapped both error messages and cleaned up the error checks.